src/entities/supportRequest.ts

```diff
@@ -0,0 +1,126 @@
+import { DateTime } from 'luxon';
+import { Entity, PrimaryGeneratedColumn, Column, BaseEntity, BeforeInsert, CreateDateColumn, UpdateDateColumn } from 'typeorm';
+import logger from '../logger';
+
+export enum SupportRequestStatus {
+  Pending = 'Pending',
+  InProgress = 'InProgress',
+  Complete = 'Complete',
+  Abandoned = 'Abandoned',
+}
+
+export enum SupportRequestType {
+  IdeaPitch = 'IdeaPitch',
+  TechnicalSupport = 'TechnicalSupport',
+}
+
+export enum SupportRequestErrors {
+  ExistingActiveRequest = 'ExistingActiveRequest',
+}
+
+@Entity()
+export class SupportRequest extends BaseEntity {
+  constructor(slackIdId: string, name: string, type: SupportRequestType) {
+    super();
+
+    this.slackId = slackIdId;
+    this.name = name;
+    this.type = type;
+  }
+
+  @PrimaryGeneratedColumn()
+  id: number;
+
+  @CreateDateColumn()
+  createdAt: Date;
+
+  @UpdateDateColumn()
+  updatedAt: Date;
+
+  @Column({ nullable: true })
+  movedToInProgressAt: Date;
+
+  @Column({ nullable: false })
+  slackId: string;
+
+  @Column({ nullable: false })
+  name: string;
+
+  @Column({ nullable: false, default: SupportRequestStatus.Pending, type: 'simple-enum', enum: SupportRequestStatus })
+  status: SupportRequestStatus;
+
+  @Column({ nullable: false, type: 'simple-enum', enum: SupportRequestType })
+  type: SupportRequestType;
+
+  @BeforeInsert()
+  async checkForCurrentRequest(): Promise<void> {
```

Comment:
Can we rename this to something that is more clear that it fails if there are active requests? Maybe the following?
```suggestion
  async validateNoActiveRequests(): Promise<void> {
```